llvm/include/llvm/MC/MCAsmBackend.h

```diff
@@ -101,7 +101,8 @@ class MCAsmBackend {
   /// Hook to check if a relocation is needed for some target specific reason.
   virtual bool shouldForceRelocation(const MCAssembler &Asm,
                                      const MCFixup &Fixup,
-                                     const MCValue &Target) {
+                                     const MCValue &Target,
+                                     const MCSubtargetInfo *STI) {
```

Comment:
Should this really be allowed to be null? Should it be a reference?

Reply:
Where `handleFixup` is called in `layout` can pass nullptr since not all fragments have a subtarget.

Reply:
Doesn't the STI argument shadow the class members of derived classes?  For example  https://github.com/llvm/llvm-project/blob/859338a695169d2804d6999f44e1e94bedf6ddd7/llvm/lib/Target/RISCV/MCTargetDesc/RISCVAsmBackend.h#L25? Looking at the RISCV case where `STI.hasFeature(RISCV::FeatureRelax)` changes to `STI->hasFeature(RISCV::FeatureRelax)`  I think it would easy to make a mistake an select the wrong one ...

I don't see an easy way to handle that aside from renaming either the parameter or class member.

Reply:
It does shadow. The compiler did complain that `.` was used on a pointer. So the compiler won't let you get to the member. There are already other instances of shadowing in this file.

Reply:
Thanks for the clarification, and yeah, we tend to shadow quite a lot I'm noticing, so you can ignore that concern.